edgelet/iotedge/src/check/checks/well_formed_connection_string.rs

```diff
@@ -49,10 +49,18 @@ impl WellFormedConnectionString {
             check.iothub_hostname = Some(hub);
             self.iothub_hostname = check.iothub_hostname.clone();
         } else if check.iothub_hostname.is_none() {
-            let warning = "Device not configured with manual provisioning, in this configuration 'iotedge check' is not able to discover the device's backing IoT Hub.\n\
-                            To run connectivity checks in this configuration please specify the backing IoT Hub name using --iothub-hostname switch if you have that information.\n\
-                            If no hostname is provided, all hub connectivity tests will be skipped.";
-            return Ok(CheckResult::Warning(Context::new(warning).into()));
+            let provisioning_file_path = settings
+                .homedir()
+                .join("cache")
+                .join("provisioning_backup.json");
+
+            let provision_result = provisioning::restore(provisioning_file_path).map_err(|e| {
+                let reason = format!("Couldn't restore provisioning file: {}", e);
```

Comment:
Might be useful to include the message of the original warning still, but change it to also mention that we tried to use the provisioning backup but that also failed. You are still honoring the `--iothub-hostname` parameter after all, so you might as well still let people know it exists in the message. But do keep it as an error instead of a warning.

Check with Venkat for the exact verbiage.

Reply:
ok, applied venkat's feedback